src/giskardpy/python_interface.py

```diff
@@ -480,12 +480,28 @@ def set_json_goal(self,
         for k, v in kwargs.copy().items():
             if v is None:
                 del kwargs[k]
-            if isinstance(v, Message):
-                kwargs[k] = convert_ros_message_to_dictionary(v)
+            else:
+                kwargs[k] = self.parse_messages(v)
         kwargs = replace_prefix_name_with_str(kwargs)
         constraint.parameter_value_pair = json.dumps(kwargs)
         self.cmd_seq[-1].constraints.append(constraint)
 
+    def parse_messages(self,
```

Comment:
why is this a `self` function? from what I can tell, it doesn't need self. You should probably modify `convert_ros_message_to_dictionary` instead

Reply:
yes correct, this functionality is now moved into `convert_ros_message_to_dictionary`